com.visualstudio.code.json

```diff
@@ -0,0 +1,125 @@
+            },
+            {
+                "type": "shell",
+                "commands": ["autoreconf -f"]
+            }
+        ]
+    },
+    {
+        "name": "xkbfile",
+        "cleanup": ["/include", "/lib/*.la", "/lib/pkgconfig"],
+        "config-opts": ["--disable-static"],
+        "sources": [
+        {
+            "type": "archive",
+            "url": "https://www.x.org/releases/individual/lib/libxkbfile-1.0.9.tar.bz2",
+            "sha256": "51817e0530961975d9513b773960b4edd275f7d5c72293d5a151ed4f42aeb16a"
+        }
+        ]
+    },
+    {
+        "name": "git",
+        "make-args": ["NO_PERL=1"],
+        "make-install-args": ["NO_PERL=1"],
+        "sources": [
+            {
+                "type": "archive",
+                "url": "https://www.kernel.org/pub/software/scm/git/git-2.14.1.tar.xz",
+                "sha256": "6f724c6d0e9e13114ab35db6f67e1b2c1934b641e89366e6a0e37618231f2cc6"
+            }
+        ]
+    },
+    {
+        "name": "nodejs",
+        "sources": [
+            {
+                "type": "archive",
+                "url": "https://nodejs.org/dist/v7.9.0/node-v7.9.0.tar.xz",
+                "sha256": "a569764b884929f31a0772600a5cf36048ad42aa8c54ba4cabea6c067ef96a29"
+            }
+        ]
+    },
+    {
+        "name": "vscode",
+        "buildsystem": "simple",
+        "build-commands": [
+            "install apply_extra /app/bin",
+            "install code.sh /app/bin/code",
+            "cp /usr/bin/ar /app/bin"
+        ],
+        "cleanup": [
+            "/share/lintian",
+            "/share/pixmaps"
+        ],
+        "sources": [
+            {
+                "type": "script",
+                "dest-filename": "apply_extra",
+                "commands": [
+                    "ar x code.deb",
+                    "rm -f code.deb",                    
+                    "tar -xf data.tar.xz",
+                    "rm -f control.tar.gz data.tar.gz debian-binary",                    
+                    "mv usr/* .",
+                    "rmdir usr",
+                    "mkdir -p export/share/icons/hicolor/512x512/apps",
+                    "cp share/pixmaps/code.png export/share/icons/hicolor/512x512/apps/com.visualstudio.code.png",
+                    "mkdir -p export/share/applications",
+                    "cp share/applications/code.desktop export/share/applications/com.visualstudio.code.desktop",                
+                    "desktop-file-edit --set-key=Exec --set-value='code %F' export/share/applications/com.visualstudio.code.desktop",
+                    "desktop-file-edit --set-key=Icon --set-value='com.visualstudio.code' export/share/applications/com.visualstudio.code.desktop",                    
+                    "mkdir -p export/share/appdata",
+                    "sed s/code.desktop/com.visualstudio.code.desktop/ share/appdata/code.appdata.xml > export/share/appdata/com.visualstudio.code.appdata.xml"        
```

Comment:
The license of this is incorrect: `<project_license>Multiple, see https://code.visualstudio.com/license</project_license>`

It should be `LicenseRef-proprietary=https://code.visualstudio.com/license`

Reply:
Also its valuable for us to add `<releases>`.

Reply:
I mean, this is upstream appdata. I'm always reluctant to override that where not actually necessary - but fair enough on the proprietary thing.

I'll open an issue upstream on this and the <releases> front, they've seemed pretty responsive to suggestions and patches thus far and I don't want to patch that downstream if possible since it's a bit fragile with a frequently released app..

Reply:
Yea, the license fix is required but for releases reasonable to get upstream to do it.